tests/unit/test.config.js

```diff
@@ -0,0 +1,162 @@
+import path from 'path';
+
+import {assert} from 'chai';
+import {describe, it} from 'mocha';
+import sinon from 'sinon';
+import {fs} from 'mz';
+
+//import {fake} from './helpers';
+import {Program} from '../../src/program';
+import {
+  applyConfigToArgv,
+  parseConfig,
+  loadJSConfigFile,
+} from '../../src/config';
+import {withTempDir} from '../../src/util/temp-dir';
+import {UsageError} from '../../src/errors';
+
+function makeArgv({
+  userCmd = [],
+  command = 'fakecommand',
+  commandDesc = 'this is a fake command',
+  commandExecutor = sinon.stub(),
+  commandOpt,
+  globalOpt,
+}) {
+  const program = new Program(userCmd);
+
+  if (globalOpt) {
+    program.setGlobalOptions(globalOpt);
+  }
+  if (commandOpt) {
+    program.command(command, commandDesc, commandExecutor, commandOpt);
+  }
+  return program;
```

Comment:
it looks like you can return an object with commonly needed properties for convenience:
````js
return {
  argv: program.yargs.exitProcess(false).argv,
  defaultValues: program.defaultValues,
};
````

And use them in tests like:
````js
const {argv, defaultValues} = makeArgv(...);
````